djinni/cwrapper/wrapper_marshal.hpp

```diff
@@ -111,34 +111,34 @@ std::unique_ptr<DjinniType> Primitive<CppType, DjinniType>::newOptional(CppType
 
 } // namespace djinni::optional
 
-// Throw in cpp exception coming from Python
+// Throw in cpp exception coming from front-end
 void cw_throw_if_pending();
-void cw_throw_exception(Handle<DjinniPythonExceptionHandle> e_handle); // weak
-void cw_default_throw_exception(djinni::Handle<DjinniPythonExceptionHandle> e_handle);
+void cw_throw_exception(Handle<DjinniFrontEndExceptionHandle> e_handle); // weak
+void cw_default_throw_exception(djinni::Handle<DjinniFrontEndExceptionHandle> e_handle);
 
-// Register in the exception state a handle to the current exception that occured in either python or cpp user code
-// to allow Python to retrieve the exception
+// Register in the exception state a handle to the current exception that occured in either
+// front-end or cpp user code to allow the front-end to retrieve the exception
 void cw_set_pending_exception(const std::exception_ptr & e_ptr);
-djinni::Handle<DjinniPythonExceptionHandle> cw_get_py_exception(const std::exception_ptr & e_ptr) noexcept; // weak
-djinni::Handle<DjinniPythonExceptionHandle> cw_default_get_py_exception(const std::exception_ptr & e_ptr) noexcept;
+djinni::Handle<DjinniFrontEndExceptionHandle> cw_get_front_end_exception(const std::exception_ptr & e_ptr) noexcept; // weak
+djinni::Handle<DjinniFrontEndExceptionHandle> cw_default_get_front_end_exception(const std::exception_ptr & e_ptr) noexcept;
 
 #define CW_TRANSLATE_EXCEPTIONS_RETURN(ret) \
     catch (const std::exception &) { \
         djinni::cw_set_pending_exception(std::current_exception()); \
         return ret; \
     }
 
-class py_exception: public std::exception {
+class front_end_exception: public std::exception {
```

Comment:
I'm not 100% sold on this one. I'd prefer not specifying "front_end" here but I found that the intent becomes confusing with the exception that's thrown from C++.

Reply:
we can not call it just `exception` either 
what about `djinni_exception` ?

or just leave it as it is, this is kind of internal anyway